cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResources.java

```diff
@@ -0,0 +1,98 @@
+package com.sequenceiq.cloudbreak.cloud.azure;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.microsoft.azure.management.compute.implementation.DiskEncryptionSetInner;
+import com.microsoft.azure.management.resources.ResourceGroup;
+import com.sequenceiq.cloudbreak.cloud.EncryptionResources;
+import com.sequenceiq.cloudbreak.cloud.azure.client.AzureClient;
+import com.sequenceiq.cloudbreak.cloud.azure.client.AzureClientService;
+import com.sequenceiq.cloudbreak.cloud.model.Platform;
+import com.sequenceiq.cloudbreak.cloud.model.Variant;
+import com.sequenceiq.cloudbreak.cloud.model.encryption.CreatedDiskEncryptionSet;
+import com.sequenceiq.cloudbreak.cloud.model.encryption.DiskEncryptionSetCreationRequest;
+
+@Service
+public class AzureEncryptionResources implements EncryptionResources {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureEncryptionResources.class);
+
+    private static final Pattern ENCRYPTION_KEY_URL_PATTERN = Pattern.compile("https://([^.]+)\\\\.vault.*");
+
+    @Inject
+    private AzureClientService azureClientService;
+
+    @Inject
+    private AzureUtils azureUtils;
+
+    @Override
+    public Platform platform() {
+        return AzureConstants.PLATFORM;
+    }
+
+    @Override
+    public Variant variant() {
+        return AzureConstants.VARIANT;
+    }
+
+    @Override
+    public CreatedDiskEncryptionSet createDiskEncryptionSet(DiskEncryptionSetCreationRequest diskEncryptionSetCreationRequest) {
+        AzureClient azureClient = azureClientService.getClient(diskEncryptionSetCreationRequest.getCloudCredential());
+        String encryptionKeyUrl = diskEncryptionSetCreationRequest.getEncryptionKeyUrl();
+        String envName = diskEncryptionSetCreationRequest.getEnvName();
+        String location = diskEncryptionSetCreationRequest.getRegion().getRegionName();
+        Map<String, String> tags = diskEncryptionSetCreationRequest.getTags();
+        String vaultName = ENCRYPTION_KEY_URL_PATTERN.matcher(encryptionKeyUrl).group(1);
+
+        String desResourceGroupName = null;
+        String vaultResourceGroupName = null;
+        if (diskEncryptionSetCreationRequest.isSingleResourceGroup()) {
+            desResourceGroupName = diskEncryptionSetCreationRequest.getResourceGroup();
+            vaultResourceGroupName = desResourceGroupName;
+        } else {
+            desResourceGroupName = (String) createResourceGroup(azureClient, envName, location, tags);
+            vaultResourceGroupName = azureClient.getVaultResourceGroup(vaultName);
+        }
+        String sourceVaultId = String.format("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.KeyVault/vaults/%s",
+                azureClient.getAzure().subscriptionId(), vaultResourceGroupName, vaultName);
```

Comment:
As mentioned in my comment in `AzureClient.getVaultResourceGroup()`, the current logic there possibly returns the whole Resource ID, so there is no need to do this magic here. In fact, if the output of that method is indeed the Resource ID, the string constructed here will be invalid. Could you please verify how this works locally?